src/builtins/roc/Bool.roc

```diff
@@ -0,0 +1,76 @@
+module [Bool, Eq, true, false, not, is_eq, is_not_eq]
+
+## Defines a type that can be compared for total equality.
+##
+## Total equality means that all values of the type can be compared to each
+## other, and two values `a`, `b` are identical if and only if `is_eq(a, b)` is
+## `Bool.true`.
+##
+## Not all types support total equality. For example, [`F32`](../Num#F32) and [`F64`](../Num#F64) can
+## be a `NaN` ([Not a Number](https://en.wikipedia.org/wiki/NaN)), and the
+## [IEEE-754](https://en.wikipedia.org/wiki/IEEE_754) floating point standard
+## specifies that two `NaN`s are not equal.
+Eq a : a where
+        ## Returns `Bool.true` if the input values are equal. This is
+        ## equivalent to the logic
+        ## [XNOR](https://en.wikipedia.org/wiki/Logical_equality) gate. The infix
+        ## operator `==` can be used as shorthand for `Bool.is_eq`.
+        ##
+        ## **Note** that when `is_eq` is determined by the Roc compiler, values are
+        ## compared using structural equality. The rules for this are as follows:
+        ##
+        ## 1. Tags are equal if their name and also contents are equal.
+        ## 2. Records are equal if their fields are equal.
+        ## 3. The collections [Str], [List], [Dict], and [Set] are equal iff they
+        ## are the same length and their elements are equal.
+        ## 4. [Num] values are equal if their numbers are equal. However, if both
+        ## inputs are *NaN* then `is_eq` returns `Bool.false`. Refer to `Num.is_nan`
+        ## for more detail.
+        ## 5. Functions cannot be compared for structural equality, therefore Roc
+        ## cannot derive `is_eq` for types that contain functions.
+        a.is_eq(a) -> Bool
+
+## Represents the boolean true and false using an opaque type.
```

Comment:
I think we're going with `nominal` type instead of opaque now

Reply:
Yeah I did NO work on reforming the comments. This is a straight copy from the builtins in the Rust compilers